tasks/DependencyCheck/task/utility.ts

```diff
@@ -5,48 +5,72 @@ import emoji = require('node-emoji');
 import tl = require('azure-pipelines-task-lib/task');
 import http = require('https');
 import path = require('path');
+import { ContainerClient, AnonymousCredential } from "@azure/storage-blob";
 
 export function cleanDependencyCheckData(): void {
   const p = path.join(__dirname, 'dependency-check-cli', 'data');
   try {
     tl.checkPath(p, 'Dependency check cli data folder');
     tl.rmRF(p);
   } catch (e) {
-    tl.debug(`An error was caugh during cleanup ${e}`);
+    tl.debug(`An error was caught during cleanup ${e}`);
     tl.warning(`Data path did not exist. The task will attempt to create it at: ${p}`);
   }
 
   tl.mkdirP(p);
 }
 
-export async function getVulnData(databaseEndpoint: string, blobName: string, filePath: string): Promise<void> {
+export async function downloadVulnData(readStorageAccountContainerSasUri: string, filePath: string, taskVersion: string): Promise<void> {
   const file = fs.createWriteStream(filePath);
-  const vulnUrl = `${databaseEndpoint}/${blobName}`;
+  const anonymousCredential = new AnonymousCredential();
+  const blobServiceClient = new ContainerClient(
+    `${readStorageAccountContainerSasUri}`,
+    anonymousCredential
+  );
+  const blobName = `${taskVersion}/${path.basename(filePath)}`;
+  const blockBlobClient = blobServiceClient.getBlockBlobClient(blobName);
   return new Promise<void>((resolve, reject) => {
-    http.get(vulnUrl, (response: any) => {
-      response.pipe(file);
-      console.log(`${emoji.get('timer_clock')}  Downloading file [${vulnUrl}]`);
-      file.on('finish', () => {
-        file.close();
-        console.log(`${emoji.get('heavy_check_mark')}  File download complete!`);
-        resolve();
-      });
+    try {
+      blockBlobClient.downloadToFile(filePath)
+        .then(() => console.log(`Successful download of vulnerability data file ${blobName}`))
+        .catch((e) => reject(new Error(`Download of vulnerability data file ${blobName} failed with error code: ${e.message}`)))
+        .finally(resolve)
+    }
+    catch (e) {
+      reject(new Error(`Download of vulnerability data file ${blobName} failed with error code: ${e.message}`));
+    }
+  });
+}
 
-      file.on('error', (err) => {
-        fs.unlink(filePath, () => { });
-        reject(new Error(err));
-      });
-    });
+export async function UploadVulnData(writeStorageAccountContainerSasUri: string, filePath: string, taskVersion: string): Promise<void> {
```

Comment:
casing on UploadVulnData is inconsistent, other function names are camal cased

Reply:
Updated to camelCase